pkg/apis/azure/validation/infrastructure.go

```diff
@@ -116,5 +126,11 @@ func ValidateInfrastructureConfigUpdate(oldConfig, newConfig *apisazure.Infrastr
 		allErrs = append(allErrs, field.Forbidden(fldPath.Child("zoned"), "moving a zoned cluster to a non-zoned cluster is not allowed"))
 	}
 
+	if newConfig.Networks.NatGateway != nil &&
+		newConfig.Networks.NatGateway.IdleConnectionTimeoutMinutes != nil &&
+		(*newConfig.Networks.NatGateway.IdleConnectionTimeoutMinutes < natGatewayMinTimeoutValue || *newConfig.Networks.NatGateway.IdleConnectionTimeoutMinutes > natGatewayMaxTimeoutValue) {
+		allErrs = append(allErrs, field.Invalid(fldPath.Child("networks", "natGateway", "IdleConnectionTimeoutMinutes"), *newConfig.Networks.NatGateway.IdleConnectionTimeoutMinutes, "idleConnectionTimeoutMinutes values must range between 4 and 120"))
+	}
+
```

Comment:
You don't have to validate it here again because `ValidateInfrastructureConfig` is also called on updates https://github.com/gardener/gardener-extension-provider-azure/blob/b3913d8541bd12c4afc0bf51cfdffa2ec21430a1/pkg/validator/shoot_validator.go#L77.